File: EXAMEN VIERNES/sistema-reserva-libros/src/main/java/com/diegoAlvarez/sistema_reserva_libros/service/UsuarioService.java
package com.diegoAlvarez.sistema_reserva_libros.service;

import com.diegoAlvarez.sistema_reserva_libros.dto.UsuarioDTO;
import com.diegoAlvarez.sistema_reserva_libros.model.Usuario;

import java.util.List;

public interface UsuarioService {
    Usuario crearUsuario(UsuarioDTO usuarioDTO);
    List<Usuario> listarUsuarios();
    Usuario obtenerUsuarioPorId(Long id);
}

File: EXAMEN VIERNES/FRONTEND/sistema-reserva-libros-frontend/src/app/services/usuario.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ApiResponse } from '../models/api-response.model';
import { Usuario, UsuarioDTO } from '../models/usuario.model';

@Injectable({
  providedIn: 'root'
})
export class UsuarioService {
  private readonly url = 'http://localhost:8080/api/usuarios';

  constructor(private readonly http: HttpClient) { }

  listarUsuarios(): Observable<ApiResponse<Usuario[]>> {
    return this.http.get<ApiResponse<Usuario[]>>(this.url);
  }

  crearUsuario(usuario: UsuarioDTO): Observable<ApiResponse<Usuario>> {
    return this.http.post<ApiResponse<Usuario>>(this.url, usuario);
  }
}